receiver/otlpreceiver/otlp.go

```diff
@@ -169,3 +173,40 @@ func (r *otlpReceiver) registerLogsConsumer(ctx context.Context, tc consumer.Log
 	}
 	return nil
 }
+
+// httpBodyCompressionHandler is a HTTP middleware checks the "Content-Encoding" HTTP header
+// and if a compression such as "gzip", "deflate", "zlib", is found,
+// the request body will be uncompressed accordingly.
+// It it encounters an error while do is, the original request is passed untouched.
+//
+// This was added to wrap the router generated by grpc-gateway as it does not provide a mechanism
+// to do this natively.
+func httpBodyCompressionHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.Header.Get("Content-Encoding") {
+		case "gzip":
+			gr, err := gzip.NewReader(r.Body)
+			if err != nil {
+				break
+			}
+			defer gr.Close()
+			body, err := ioutil.ReadAll(gr)
+			if err != nil {
+				break
+			}
+			r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
```

Comment:
Can this be done in a streaming fashion instead of allocating an entire buffer in memory?